+BE_Controller/Evaluation.m

```diff
@@ -304,30 +316,84 @@ function evaluate(view, model)
+                            r = [{'Lower'}, {'Upper'}];
+                            % Calculate Brillouin shift range
+                            for s_ind = 1:length(s)
+                                for r_ind = 1:length(r)
+                                    val = constraints.(s{s_ind}).(r{r_ind});
+                                    if isnumeric(val)
+                                        % If the value is not Inf, we have to
+                                        % convert from GHz to pix
+                                        if ~isinf(val)
+                                            valid = ~isnan(calibration.frequency);
+                                            if ~isempty(calibration.frequency) && sum(valid(:))
+                                                x = BE_SharedFunctions.getFrequencyFromMap(1:length(spectrum), time, calibration);
+
+                                                ind = interp1(x, 1:length(x), f_Rayleigh - val);
+                                                constraints.(s{s_ind}).(r{r_ind}) = ind - min(ind_Brillouin_shifted(:)) + 1;
+                                            end
+                                        end
+                                    else
+                                        switch lower(val)
+                                            case 'min'
+                                                if invert
+                                                    constraints.(s{s_ind}).(r{r_ind}) = length(BrillouinSection);
+                                                else
+                                                    constraints.(s{s_ind}).(r{r_ind}) = 1;
+                                                end
+                                            case 'max'
+                                                if invert
+                                                    constraints.(s{s_ind}).(r{r_ind}) = 1;
+                                                else
+                                                    constraints.(s{s_ind}).(r{r_ind}) = length(BrillouinSection);
+                                                end
+                                            otherwise
+                                                constraints.(s{s_ind}).(r{r_ind}) = NaN;
+                                        end
+                                    end
+                                end
+                                % If the Upper and Lower bound are
+                                % switched, correct it
+                                if constraints.(s{s_ind}).Lower > constraints.(s{s_ind}).Upper
+                                    tmp = constraints.(s{s_ind}).Lower;
+                                    constraints.(s{s_ind}).Lower = constraints.(s{s_ind}).Upper;
+                                    constraints.(s{s_ind}).Upper = tmp;
+                                end
+                            end
```

Comment:
Move this block https://github.com/BrillouinMicroscopy/BrillouinEvaluation/pull/14/files#diff-d1c547a949edd3cb5ab68953fb172c78R319-R375 to line 388 to only calculate the constraints if necessary.

Reply:
Done with d2e56f69cb6eb93d179283f1bf92c4b6532353b8